cmd/kubeadm/app/images/images_test.go

```diff
@@ -217,15 +218,47 @@ func TestGetAllImages(t *testing.T) {
 			},
 			expect: constants.Etcd,
 		},
+		// add-ons' images should follow ClusterConfiguration.AddOns
 		{
-			name:   "CoreDNS image is returned",
-			cfg:    &kubeadmapi.ClusterConfiguration{},
+			name: "CoreDNS image is returned if set to true",
+			cfg: &kubeadmapi.ClusterConfiguration{
+				AddOns: map[string]bool{
+					"coredns": true,
+				},
+			},
 			expect: constants.CoreDNSImageName,
 		},
+		{
+			name: "KubeProxy image is returned if set to true",
+			cfg: &kubeadmapi.ClusterConfiguration{
+				AddOns: map[string]bool{
+					"kube-proxy": true,
+				},
+			},
+			expect: constants.KubeProxy,
+		},
+		{
+			name: "CoreDNS and KubeProxy images should be skipped when set to false",
+			cfg: &kubeadmapi.ClusterConfiguration{
+				AddOns: map[string]bool{
+					"coredns":    false,
+					"kube-proxy": false,
+				},
+			},
+			expect:     "",
+			expectNots: []string{constants.CoreDNSImageName, constants.KubeProxy},
+		},
 	}
 	for _, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
 			imgs := GetControlPlaneImages(tc.cfg)
+			for _, img := range imgs {
+				for _, expectNot := range tc.expectNots {
+					if strings.Contains(img, expectNot) {
+						t.Fatalf("shouldn't find %q in %q", expectNot, imgs)
+					}
+				}
+			}
 			for _, img := range imgs {
 				if strings.Contains(img, tc.expect) {
 					return
```

Comment:
those logic should be updated, `strings.Contains(img, "")` will already return true.

Reply:
Thanks ! Done.